spec/support/matchers/flash_matchers.rb

```diff
@@ -0,0 +1,11 @@
+RSpec::Matchers.define :have_flash_message do |flash_type|
+  match do |page|
+    if flash_type.present?
+      page.has_css?("div.flash_#{flash_type}")
+      within("div.flashes") { page.has_text }
```

Comment:
Thanks. That make sense and will definitely help me make better use of Capybara's finders.